File: 20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        lst=["(","{","["]
        for i in s:
            if i in lst:
                stack.append(i)
            if stack and i=='}' and stack[-1]=="{":
                stack.pop()
            elif stack and i==']'and stack[-1]=="[":
                stack.pop()
            elif stack and i==')'and stack[-1]=="(":
                stack.pop()
            elif i==")" or i=="}" or i=="]":
                stack.append(i)
        return len(stack)==0

File: 20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>stk;
        char v;
        for(int i=0;i<s.size();i++){
            if(s[i]=='[' || s[i]=='{' || s[i]=='(') stk.push(s[i]);
            else if(!stk.empty()){
                v=stk.top();
                if((s[i]==')' && v=='(') || (s[i]==']' && v=='[' ) || (s[i]=='}' && v=='{'))
                    stk.pop();
                else return false;
            }
            else if(s[i]==']' || s[i]=='}' || s[i]==')') stk.push(s[i]);
        }
        return stk.size()==0;
    }
};
